driver-core/src/main/com/mongodb/ReadPreferenceHedgeOptions.java

```diff
@@ -0,0 +1,120 @@
+/*
+ * Copyright 2008-present MongoDB, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.mongodb;
+
+import com.mongodb.annotations.Immutable;
+import org.bson.BsonBoolean;
+import org.bson.BsonDocument;
+
+/**
+ * Options to apply to hedged reads in the server.
+ *
+ * @since 4.1
+ * @mongodb.server.release 4.4
+ */
+@Immutable
```

Comment:
I noticed that `TransactionOptions` is immutable, but `BulkWriteOptions` is not. What is the preferred convention the Options classes?

Reply:
This one should be immutable, because it's reachable from something that should be immutable: MongoClientSettings.  We don't worry about mutability of options classes intended to be used on a per-operation basis (e.g. UpdateOptions)

Reply:
Though perhaps we can avoid the builder by using the same pattern as we are now using for `ReadPreference` itself: use `with` style modifiers in the future if we ever add new fields to this class.  So:

      ReadPreferenceHedgeOptions.enabled()
      ReadPreferenceHedgeOptions.disabled()

and in the future:

      ReadPreferenceHedgeOptions.enabled().withDelay(100, TimeUnit.MILLISECONDS)

What do you think?

Reply:
I think the builder is fine if the class is immutable. It was just interesting to note that the only immutable options classes we have are: `TransactionOptions`, and `MongoClientOptions` and both use the Builder pattern.  So I think we should stick with the builder for now.